learning/modules/quantum-mechanics/getting-started-with-qiskit.ipynb

```diff
@@ -0,0 +1,1291 @@
+    "Remember, $N$ bits allows you to represent the numbers $0$ through $2^N - 1$, so using 6 bits would get us up to $2^6 - 1 = 63$. That's not quite enough. We add one more bit to get up to $2^7 - 1 = 127$. Now lets break 86 down into powers of 2:\n",
+    "\n",
+    "$$\n",
+    "\\begin{aligned}\n",
+    "86 &= 64 + 16 + 4 + 2 \\\\\n",
+    "&= 2^6 \\times 1 + 2^5 \\times 0 + 2^4 \\times 1 + 2^3 \\times 0 + 2^2 \\times 1 + 2^1 \\times 1 + 2^0 \\times 0 \\\\\n",
+    "&= 1010110\n",
+    "\\end{aligned}\n",
+    "$$\n",
+    "</details>\n",
+    "\n",
+    "\n",
+    "### Fundamental operations: gates\n",
+    "\n",
+    "Now, a computer needs to be able to *do* something with the bits in order to, well, compute. Binary gates are the operations that form the fundamental building blocks of all more complicated algorithms and codes.\n",
+    "\n",
+    "### Single-bit gate:\n",
+    "\n",
+    "#### NOT\n",
+    "\n",
+    "When you have only one bit, there's only one way you can transform its state: flip the state from 0 to 1 or from 1 to 0. We call this the \"NOT\" gate. The effect of this gate — and the rest of the gates we'll discuss below — can be represented in a so-called \"truth table,\" with columns for the input and output states of the qubits. The truth table for the NOT gate is:\n",
+    "\n",
+    "| Input | Output |\n",
+    "| :------- | -------: |\n",
+    "| 0      |1  |\n",
+    "| 1  | 0  |\n",
+    "\n",
+    "### Multi-bit gates:\n",
+    "\n",
+    "#### AND\n",
+    "\n",
+    "AND is a two-bit gate that takes two input bits and outputs a single bit. It outputs 1 if both input bits are 1, and 0 otherwise:\n",
+    "\n",
+    "| Input | Output |\n",
+    "| :------- | -------: |\n",
+    "| 00      |0  |\n",
+    "| 01  | 0  |\n",
+    "| 10      |0  |\n",
+    "| 11  | 1  |\n",
+    "\n",
+    "#### OR\n",
+    "\n",
+    "OR is another two-bit gate with a single output bit. It outputs 1 if *either* of the bits are 1:\n",
+    "\n",
+    "| Input | Output |\n",
+    "| :------- | -------: |\n",
+    "| 00      |0  |\n",
+    "| 01  | 1  |\n",
+    "| 10      |1  |\n",
+    "| 11  | 1  |\n",
+    "\n",
+    "#### XOR\n",
+    "\n",
+    "XOR stands for \"exclusive OR\" and it is like the OR gate, but outputs 1 if *only one* of the input bits are 1. It outputs 0 if they are either both 1 or both 0:\n",
+    "\n",
+    "| Input | Output |\n",
+    "| :------- | -------: |\n",
+    "| 00      | 0  |\n",
+    "| 01  | 1  |\n",
+    "| 10      | 1  |\n",
+    "| 11  | 0  |\n",
+    "\n",
+    "\n",
+    "### Measurements:\n",
+    "\n",
+    "Typically, when learning about classical computing, not much attention is paid to the process of reading out the state of the bits. This is because there's not much to it from a conceptual perspective. You can measure the bits at anytime before, during, or after a computation, and it doesn't affect the outcome. This is not the case in quantum computing, as we will discuss below.\n",
```

Comment:
```suggestion
    "Typically, when learning about classical computing, not much attention is paid to the process of reading out the state of the bits. This is because it's not very complex from a conceptual perspective. You can measure the bits at anytime before, during, or after a computation, and it doesn't affect the outcome. This is not the case in quantum computing, as we will discuss below.\n",
```